cpp/CMakeLists.txt

```diff
@@ -113,6 +115,7 @@ endif()
 
 # add parts of the project
 include(thirdparty/CMakeLists.txt)
+add_subdirectory(memilio/ad)

```

Comment:
No, the AD examples also rely on it. ad is a header-only-library. the overhead on including it should be minimal.

Reply:
adding it always is fine, but shouldn't it be in thirdparty?

Reply:
It was the explicit request of the author of the library to add the library as part of the memilio repository. The reason is that we have a tailored Apache-2.0 license especially for memilio. Apart from that, the AD library is released under GPL-3 which is not compatible with Apache-2.0. To minimize the risk that someone else extracts the AD-library under the Apache-2.0 license from memilio it should not be placed under third party.

Reply:
to be clear, it would still be in the repository, just a different folder.

Reply:
How hard would it be to replace? Not necessarily right away. I think most AD libraries work in the same way.

Reply:
Probably, it would not be too difficult to interface another AD library based on overloading. However, the current library is quite performant and we also get support on short notice if we need. However, in then license issues have to be considered.

Reply:
It's decided to keep it as is.